src/module/actor/hazard/document.ts

```diff
@@ -98,6 +98,14 @@ class HazardPF2e<TParent extends TokenDocumentPF2e | null = TokenDocumentPF2e |
         details.alliance = null;
     }
 
+    /** Run rule elements */
+    override prepareEmbeddedDocuments(): void {
+        super.prepareEmbeddedDocuments();
+        for (const rule of this.rules) {
```

Comment:
Why not just move this to the base class?

Reply:
As a refactor? Sure. I'm down. I assumed there was a reason it wasn't.

Reply:
I think it was a redundant safeguard on them not getting executed by loot actors.